File: java/ex3/src/BoxedItem.java
/** Class for objects that can be sold, packed, and shipped. */
public class BoxedItem implements Sellable, Transportable {
  private String descript; // description of this item
  private int price; // list price in cents
  private int weight; // weight in grams
  private boolean haz; // true if object is hazardous
  private int height = 0; // box height in centimeters
  private int width = 0; // box width in centimeters
  private int depth = 0; // box depth in centimeters

  /** Constructor */
  public BoxedItem(String desc, int p, int w, boolean h) {
    descript = desc;
    price = p;
    weight = w;
    haz = h;
  }

  public String description() {
    return descript;
  }

  public int listPrice() {
    return price;
  }

  public int lowestPrice() {
    return price / 2;
  }

  public int weight() {
    return weight;
  }

  public boolean isHazardous() {
    return haz;
  }

  public int insuredValue() {
    return price * 2;
  }

  public void setBox(int h, int w, int d) {
    height = h;
    width = w;
    depth = d;
  }

  public static void main(String[] args) {
    // Sellable s1 = new BoxedItem("Glass", 50, 20, false);
    // System.out.println(s1.listPrice());

    // System.out.println(s1.isHazardous());

    Transportable s1 = new BoxedItem("Glass", 50, 20, false);
    System.out.println(((BoxedItem) s1).listPrice());
  }
}

File: python/ex3/BoxedItem.py
from Sellable import Sellable
from Transportable import Transportable


class BoxedItem(Sellable, Transportable):
    def __init__(self, desc, p, w, h):
        self.descript = desc  # description of this item
        self.price = p  # list price in cents
        self.weight = w  # weight in grams
        self.haz = h  # true if object is hazardous
        self.height = 0  # box height in centimeters
        self.width = 0  # box width in centimeters
        self.depth = 0  # box depth in centimeters

    def description(self):
        return self.descript

    def listPrice(self):
        return self.price

    def lowestPrice(self):
        return self.price // 2

    def weight(self):
        return self.weight

    def isHazardous(self):
        return self.haz

    def insuredValue(self):
        return self.price * 2

    def setBox(self, h, w, d):
        self.height = h
        self.width = w
        self.depth = d


def main():
    # Sellable s1 = new BoxedItem("Glass", 50, 20, false);
    # System.out.println(s1.listPrice());

    # System.out.println(s1.isHazardous());

    s1 = BoxedItem("Glass", 50, 20, False)
    print(s1.listPrice())


if __name__ == "__main__":
    main()
